### PR TITLE
MODSOURCE-202: Records stream API

### PR DESCRIPTION
**Pull request by wwelling**
resolves [MODSOURCE-202](https://issues.folio.org/browse/MODSOURCE-202)

This PR moves more toward [reactive streams](https://vertx.io/docs/vertx-reactive-streams/java/) with [RxJava](https://github.com/ReactiveX/RxJava). This provides a more fluent and flexible way to manage reactive code still allowing the use of jOOQ typed query building.

### CODE REVIEW

File: ramls/source-record-storage-stream.raml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    content: Streaming API for searching records
+
+types:
+  errors: !include raml-storage/raml-util/schemas/errors.schema
+
+traits:
+  validate: !include raml-storage/raml-util/traits/validation.raml
+  pageable:  !include raml-storage/raml-util/traits/pageable.raml
+
+/source-storage/stream:
```
Comment (KaterynaSenchenko) [inline ramls/source-record-storage-stream.raml:19]: this new endpoint should be also specified in ModuleDescriptor

**Review by KaterynaSenchenko — APPROVED** [review]

File: ramls/source-record-storage-stream.raml (hunk 1 above)
Comment (KaterynaSenchenko) [reply ramls/source-record-storage-stream.raml:19]: never mind, I see you added it in #278

**Review by RuslanLavrov — APPROVED** [review]

File: mod-source-record-storage-server/src/main/java/org/folio/dao/RecordDaoImpl.java [hunk 2]
```diff
@@ -344,6 +342,14 @@ private ReactiveClassicGenericQueryExecutor getQueryExecutor(String tenantId) {
     return postgresClientFactory.getQueryExecutor(tenantId);
   }
 
+  private PgPool getCachecPool(String tenantId) {
```
[the remark this answers is not in the document]
Comment (wwelling) [reply mod-source-record-storage-server/src/main/java/org/folio/dao/RecordDaoImpl.java:345]: Updated in fast forward PR. Thanks.

**Comment by wwelling** [comment]
closing as merged with #278

**Pull request closed without merging**